src/components/scenes/PluginViewScene.js

```diff
@@ -224,15 +238,49 @@ class PluginView extends React.Component<PluginProps, PluginState> {
     if (navState.loading) {
       return
     }
-    // TODO: improve handling of edge-ret URIs
-    if (navState.url.match(/edge-ret:\/\/plugins/)) {
+    const parsedUrl = parse(navState.url, {}, true)
+    if (parsedUrl.protocol === 'edge:' && parsedUrl.hostname === 'x-callback-url') {
+      switch (parsedUrl.pathname) {
+        case '/paymentUri':
+          this.props.abcWallet.parseUri(parsedUrl.query.uri)
+            .then(result => {
+              const info: GuiMakeSpendInfo = {
+                currencyCode: result.currencyCode,
+                nativeAmount: result.nativeAmount,
+                publicAddress: result.publicAddress
+              }
+              this.successUrl = parsedUrl.query['x-success']
+              this.bridge.makeSpendRequest(info)
+                .then(tr => {
+                  if (this.successUrl) {
+                    this._webviewOpenUrl(this.successUrl)
+                  }
+                })
+                .catch(e => {
+                  console.log(e)
+                })
+            })
+          break
+        default:
+          console.log('nothing yet')
+      }
+
+      return
+    }
+    if (parsedUrl.protocol === 'edge-ret:') {
       Actions.pop()
       return
     }
+    if (parsedUrl.origin === this.successUrl) {
+      this.bridge.navStackClear()
+      return
+    }
+
     if (!navState.canGoForward) {
       this.bridge.navStackPush(navState.url)
     } else if (!navState.canGoBack) {
       this.bridge.navStackClear()
+    } else {
     }
```

Comment:
Unnecessary else clause, but not a deal breaker

Reply:
removed